components/ephemeral_storage/ephemeral_storage_service.cc

```diff
@@ -327,6 +339,11 @@ void EphemeralStorageService::ScheduleFirstPartyStorageAreasCleanupOnStartup() {
 
 void EphemeralStorageService::CleanupFirstPartyStorageAreasOnStartup() {
   DCHECK(!context_->IsOffTheRecord());
+  Profile* profile = Profile::FromBrowserContext(context_);
+  if (!DoesProfileHaveAnyBrowserWindow(profile)) {
```

Comment:
actually, now I'm thinking: shouldn't we instead start the cleanup process timer when a regular profile window is created (or exists) for `context_`?

right now if you open a normal window with an existing incognito window, the cleanup won't run.

Reply:
From what I saw in the code there are two situation timers that trigger the cleanup.
1. At startup, there is a 5-second keep-alive timer after which the cleanup is executed(so if you are quick enough to enter the URL in the address bar, the cleanup is aborted)
2. After closing a tab, there is a 30-second grace period (this option is enabled by default) after which all data cleanup is started.

Are you running in one of these situations? Can you please add some steps to reproduce? Thanks!